File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        
        int sum =0, min =Integer.MAX_VALUE,start=0; 
        
        for(int i =0 ; i< nums.length;i++){
            sum += nums[i];            
            while(sum>=target){
                min = Math.min(min, i-start+1);
                sum -=nums[start++];
            }
        }
        
        return min==Integer.MAX_VALUE ? 0:min;
    }
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.kt
class Solution {
    fun minSubArrayLen(target: Int, nums: IntArray): Int {
        
        var res = Int.MAX_VALUE
        var left = 0
        var sum =0 
        
        for(right in nums.indices){
            sum += nums[right]
            while(sum>=target){
                res = Math.min(res, right-left+1)
                sum -=nums[left++]
            }
            
        }
               return if(res==Integer.MAX_VALUE) 0 else res
    } 
}